pypesto/objective/amici_prediction.py

```diff
@@ -0,0 +1,439 @@
+            if 1 in sensi_orders:
+                amici_sy += [rdata['sy'] for rdata in ret['rdatas']]
+
+    def _write_to_csv(self,
+                      outputs: Sequence[np.ndarray],
+                      outputs_sensi: Sequence[np.ndarray],
+                      timepoints: Sequence,
+                      output_file: str):
+        """
+        This method saves predictions from an amici model to a csv file.
+
+        Parameters
+        ----------
+        outputs:
+            List of postprocessed outputs (ndarrays)
+        outputs_sensi:
+            List of sensitivities of postprocessed outputs (ndarrays)
+        timepoints:
+            List of output timepoints (ndarrays)
+        output_file:
+            path to file/folder to which results will be written
+        """
+        def _prepare_csv_output(output_file):
+            """
+            If a csv is requested, this routine will create a folder for it,
+            with a suiting name: csv's are by default 2-dimensional, but the
+            output will have the format n_conditions x n_timepoints x n_outputs
+            For sensitivities, we even have x n_parameters. This makes it
+            necessary to create multiple files and hence, a folder of its own
+            makes sense.
+            """
+            # allow entering with names with and without file type endings
+            if '.' in output_file:
+                output_file_path, output_file_suffix = output_file.split('.')
+            else:
+                output_file_path = output_file
+                output_file_suffix = 'csv'
+
+            # parse path
+            if '/' in output_file_path:
+                tmp = output_file_path.split('/')[-1]
+            else:
+                tmp = [output_file_path,]
+
+            output_file_dummy = tmp[-1]
+            output_path = os.path.join(*tmp)
+
+            # create folder with files contianing the return values
+            if os.path.exists(output_path):
+                output_path += str(int(time() * 1000))
+                warn('Output folder already existed! Changed the name of the '
+                     'output folder by appending the unix timestampp to make '
+                     'it unique!')
+            os.mkdir(output_path)
+
+            return output_path, output_file_dummy, output_file_suffix
+
+        # process the name of the output file, create a folder
+        output_path, output_file_dummy, output_file_suffix = \
+            _prepare_csv_output(output_file)
+
+        if outputs:
+            # loop over conditions (i.e., amici edata objects)
+            for i_out, output in enumerate(outputs):
+                i_timepoints = pd.Series(name='time', data=timepoints[i_out])
+                # create filename for this condition
+                tmp_filename = os.path.join(output_path,
+                    output_file_dummy + f'_{i_out}.' + output_file_suffix)
```

Comment:
The code I wrote took explicitly care of that, by using splits etc. ;)  (see`_prepare_csv_output()`)

Reply:
Ah yep, missed that :+1: 

This set of suggestions involving `Path` avoids such splits, in case there is some OS-specific issue, and is more readable for me, but is not a significant issue for me.